### PR TITLE
Send signalr message for episode monitored flag changes

### PR DESCRIPTION
**Pull request by Taloth**
#### Database Migration
NO

#### Description
Changing monitored flags for episodes don't result in a signalr message.
This means that second browser tabs won't update, but also third-party apps using signalr won't get notified (Bazarr specifically).

note that I could do this in EpisodeModule handling the SetMonitored api calls and send broadcasts. However this means we wouldn't send updates for monitored flag changes triggered internally.

Some concerns:

- Added `GetResourceByIdForBroadcast` since I didn't want the series to be included in the notification. I have to check if that causes problems in the UI.
- It sends out 4 notifications for the calendar, episode, cutoff, missing endpoints. Which is a bit excessive.
- I do wonder what other things lack notifications to the UI.

### CODE REVIEW

File: src/NzbDrone.Core/Tv/EpisodeRepository.cs [hunk 1]
```diff
@@ -173,25 +175,44 @@ public void SetMonitoredBySeason(int seriesId, int seasonNumber, bool monitored)
...
+            mapper.AddParameter("seriesId", seriesId);
+            mapper.AddParameter("seasonNumber", seasonNumber);
+            mapper.AddParameter("monitored", monitored);
+
+            var sqlUpdate = $"UPDATE Episodes SET Monitored = @monitored WHERE {sqlWhere}";
 
-            mapper.ExecuteNonQuery(sql);
+            mapper.ExecuteNonQuery(sqlUpdate);
+
+            episodes.ForEach(ModelUpdated);
```
Comment (markus101) [inline src/NzbDrone.Core/Tv/EpisodeRepository.cs:193]: Could avoid looping 2x and set monitored + send model updated here, but not that big of a concern.

**Review by markus101 — APPROVED** [review]
Just the one comment, I think the trade-offs are reasonable to achieve this functionality at least for now.

File: src/NzbDrone.Core/Tv/EpisodeRepository.cs (hunk 1 above)
Comment (Taloth) [reply src/NzbDrone.Core/Tv/EpisodeRepository.cs:193]: I can also just add ModelUpdate(id) coz it's gonna call GetResourceById anyway.

**Comment by Taloth** [comment]
Hmm, there's actually another problem. If I toggle a season monitored flag. It steps through the episodes surprisingly slowly.

**Pull request merged**